packages/cli-repl/src/cli-repl.ts

```diff
@@ -346,23 +353,31 @@ export class CliRepl implements MongoshIOProvider {
       },
       version
     );
+    loggingAndTelemetry.setup();
+
     markTime(TimingCategories.Telemetry, 'completed telemetry setup');
 
     if (analyticsSetupError) {
       this.bus.emit('mongosh:error', analyticsSetupError, 'analytics');
     }
 
+    // Read local and global configuration
     try {
       this.config = await this.configDirectory.generateOrReadConfig(
         this.config
       );
-    } catch (err: any) {
-      this.warnAboutInaccessibleFile(err);
+    } catch (err: unknown) {
+      this.warnAboutInaccessibleFile(err as Error);
     }
 
     this.globalConfig = await this.loadGlobalConfigFile();
     markTime(TimingCategories.UserConfigLoading, 'read global config files');
 
+    const disableLogging = this.getConfig('disableLogging');
+    if (disableLogging !== true) {
+      await this.startLogging(loggingAndTelemetry);
+    }
```

Comment:
You'll want to add an e2e test for this as well, maybe even including one that tests that the global config file can disable logging via this option